hack/build/build-go.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -e
+
+source hack/build/common.sh
+source hack/build/config.sh
+
+if [ -z $1 ]; then
+    go_opt="build"
+else
+    go_opt=$1
+    shift
+fi
+
+# If no binaries are given, assume all
+if [ $# -gt 0 ]; then
+    targets=$@
+else
+    targets=$BINARIES
+fi
+
+if [ "$go_opt" == "test" ]; then
+    if [ $# -eq 0 ]; then
+    (
+        cd ${CDI_DIR}/cmd
+        go test -v ./...
+    )
+    (
+        cd ${CDI_DIR}/pkg
+        go test -v ./...
+    )
+    (
+        cd ${CDI_DIR}/test
+        go test -v ./...
+    )
+    else
+        for tgt in ${targets}; do
+        (
+            cd $tgt
+            go test -v ./...
+        )
+        done
+    fi
+fi
+
+if [ "$go_opt" == "build" ]; then
+    for tgt in ${targets}; do
+        eval "$(go env)"
+        BIN_NAME=$(basename $tgt)
+        LINUX_NAME=${BIN_NAME}-${BUILD_OS}-${BUILD_ARCH}
+        rm -rf ${CMD_OUT_DIR}/${BIN_NAME}/${LINUX_NAME}
+        rm -f ${BIN_DIR}/${LINUX_NAME}
+        (
+            cd $tgt
+
+            go vet ./...
+
+            GOOS=${BUILD_OS} GOARCH=${BUILD_ARCH} go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${LINUX_NAME} -ldflags '-extldflags "static"'
+
+            ln -sf ${LINUX_NAME} ${CDI_DIR}/bin/${LINUX_NAME}
+        )
+    done
+fi
+
+# If no pgk paths are given, test or execute go $target for pkgs
+if [ $# -eq 0 ]; then
```

Comment:
seem like we're testing either # args or ${go_opt} several times. Is there a clean way to consolidate/DRY these checks?